Spigot-API-Patches/0299-Add-corner-getter-to-WorldBorder.patch

```diff
@@ -0,0 +1,57 @@
+From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
+From: TwoLeggedCat <80929284+TwoLeggedCat@users.noreply.github.com>
+Date: Wed, 12 May 2021 20:07:03 -0500
+Subject: [PATCH] Add corner getter to WorldBorder
+
+
+diff --git a/src/main/java/org/bukkit/Corner.java b/src/main/java/org/bukkit/Corner.java
+new file mode 100644
+index 0000000000000000000000000000000000000000..66d6edf6269e4be74e8b329df5607b136b76e548
+--- /dev/null
++++ b/src/main/java/org/bukkit/Corner.java
+@@ -0,0 +1,27 @@
++package org.bukkit;
++
++public enum Corner {
++    NORTH_EAST,
++    NORTH_WEST,
++    SOUTH_EAST,
++    SOUTH_WEST;
++
++    public boolean isNorth() {
++        switch (this) {
++            case NORTH_EAST:
++            case NORTH_WEST:
++                return true;
++            default:
++                return false;
++        }
++    }
++    public boolean isEast() {
++        switch (this) {
++            case NORTH_EAST:
++            case SOUTH_EAST:
++                return true;
++            default:
++                return false;
++        }
++    }
++}
+diff --git a/src/main/java/org/bukkit/WorldBorder.java b/src/main/java/org/bukkit/WorldBorder.java
+index afb7b136b461202026290624836446cff9f9e45d..a74776861b2fc8ef5310aa822b6dd4e4ccc39b53 100644
+--- a/src/main/java/org/bukkit/WorldBorder.java
++++ b/src/main/java/org/bukkit/WorldBorder.java
+@@ -54,6 +54,13 @@ public interface WorldBorder {
+      */
+     public void setCenter(@NotNull Location location);
+ 
++    /**
++     * Returns the specified corner of the WorldBorder
++     * @param corner Which corner to return
++     * @return Location of corner where y is always the minimum height of the world, or null if Corner is null
++     */
++    public @NotNull Location getCorner(@NotNull Corner corner);
```

Comment:
Maybe getCornerLocation would be a better name? I'm not actually sure. getCorner just sounds like its returning a `Corner` object, instead of the location of the corner. I could be overthinking that though.

Reply:
No, you are very correct about that. Will fix.